api/v1/address.go

```diff
@@ -1,55 +1,104 @@
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Addresses}
+//	@Router			/addresses [put]
 func handleSetAddresses(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+	var addresses Addresses
+	err := json.NewDecoder(c.Request().Body).Decode(&addresses)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = db.DB.Transaction(func(tx *gorm.DB) error {
+		for _, address := range addresses {
+			address.TenantID = GetTenantId(c)
+			res := tx.Where("tenant_id = ? AND address_robust in (?) AND deleted_at IS NULL").Save(&address)
+			if res.Error != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Updated Addresses associated with the tenant"))
 }
 
 // handleDeleteAddresses godoc
 //
-//		@Summary		Delete addresses used by a tenant
-//		@Security apiKey
-//	  @Param 			addresses body []string true "addresses to delete"
-//		@Produce		json
-//		@Success		200	{object}	ResponseEnvelope{response=Addresses}
-//		@Router			/addresses [delete]
+//	@Summary		Delete addresses used by a tenant
+//	@Param 			token header string true "Auth token"
+//	@Param 			addresses body []string true "addresses to delete"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Addresses}
+//	@Router			/addresses [delete]
 func handleDeleteAddresses(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+	var addressesIds []string
+	err := json.NewDecoder(c.Request().Body).Decode(&addressesIds)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = db.DB.Where("tenant_id = ? AND address_robust in (?) AND deleted_at IS NULL", GetTenantId(c), addressesIds).Delete(&Address{}).Error
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Deleted Addresses associated with the tenant"))
 }
 
 // handleGetAddresses godoc
 //
 //	@Summary		Get addresses used by a tenant
-//	@Security apiKey
+//	@Param			token header string true "Auth token"
 //	@Produce		json
 //	@Success		200	{object}	ResponseEnvelope{response=Addresses}
 //	@Router			/addresses [get]
 func handleGetAddresses(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+	var addresses []AddressResponse
+	res := db.DB.Table("addresses").Where("tenant_id = ? AND deleted_at IS NULL", GetTenantId(c)).Find(&addresses)
```

Comment:
you don't need the `deleted_at` check anywhere actually - Gorm handles this automatically and will not return them. 

as an FYI if you ever do need to get them, you can do it with the `Unscoped` method https://gorm.io/docs/delete.html#Find-soft-deleted-records